test/fs/test_fs_mkdir_dotdot.c

```diff
@@ -0,0 +1,28 @@
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include <assert.h>
+
+int main() {
+  assert(mkdir("test", 0777) == 0);
+  assert(mkdir("test/a", 0777) == 0);
+  assert(mkdir("test/a/..", 0777) == -1);
+  printf("error: %s\n", strerror(errno));
+  assert(errno == EEXIST);
+  assert(mkdir("test/a/.", 0777) == -1);
+  printf("error: %s\n", strerror(errno));
+  assert(errno == EEXIST);
+  assert(mkdir("test/a/b/..", 0777) == -1);
+  printf("error: %s\n", strerror(errno));
+  assert(errno == ENOENT);
+  assert(mkdir("test/a/b/.", 0777) == -1);
+  printf("error: %s\n", strerror(errno));
+  assert(errno == ENOENT);
+  printf("success\n");
```

Comment:
Is this worth a standalone test or is there an existing mkdir test that this can be part of?

Reply:
The two existing tests with the most mkdir calls are `test/stdio/test_rename.c` with 17 and `test/wasmfs/wasmfs_mkdir.c`  with 9. Oddly `wasmfs_mkdir` says `// FIXME: Merge with unlink test`. But maybe removing the `wasmfs` part of that and adding this to it would be a good idea?

Reply:
But `wasmfs_mkdir` can't run in rawfs due to absolute paths. Maybe do:
```C
#ifdef NODERAWFS
#def MAYBE_RELATIVE "./"
#else
#def MAYBE_RELATIVE ""
#endif
```
and prefix all absolute paths with this?